src/testsuite/vec_pwr9_dummy.c

```diff
@@ -106,6 +198,241 @@ __test_muludq_PWR9 (vui128_t *mulh, vui128_t a, vui128_t b)
+   * The high 128 bits are accumulated in t and the low 128-bits
+   * in tmq. The high 128-bits of the product are returned to the
+   * address of the 1st parm. The low 128-bits are the return
+   * value.
+   */
+  const vui64_t zero = { 0, 0 };
+  vui64_t a_swap = vec_swapd ((vui64_t) a);
+  vui128_t thq, tlq, tx;
+  vui128_t t0l, tc1;
+  vui128_t thh, thl, tlh, tll;
+  /* multiply the low 64-bits of a and b.  For PWR9 this is just
+   * vmsumudm with conditioned inputs.  */
+  tll = vec_vmuloud ((vui64_t)a, (vui64_t)b);
+  thh = vec_vmuleud ((vui64_t)a, (vui64_t)b);
+  thl = vec_vmuloud (a_swap, (vui64_t)b);
+  tlh = vec_vmuleud (a_swap, (vui64_t)b);
+  /* sum the two middle products (plus the high 64-bits of the low
+   * product.  This will generate a carry that we need to capture.  */
+  t0l   = (vui128_t) vec_mrgahd ( (vui128_t) zero, tll);
+  tc1 = vec_addcuq (thl, tlh);
+  tx   = vec_adduqm (thl, tlh);
+  tx   = vec_adduqm (tx, t0l);
+  /* result = t[l] || tll[l].  */
+  tlq = (vui128_t) vec_mrgald ((vui128_t) tx, (vui128_t) tll);
+  /* Sum the high product plus the high sum (with carry) of middle
+   * partial products.  This can't overflow.  */
+  thq = (vui128_t) vec_permdi ((vui64_t) tc1, (vui64_t) tx, 2);
+  thq = vec_adduqm ( thh, thq);
+
+  *mulu = (vui128_t) thq;
+  return ((vui128_t) tlq);
+}
+
+vui128_t
+__test_muludq_y_PWR9 (vui128_t *mulu, vui128_t a, vui128_t b)
+{
+  vui32_t t, tmq;
+  /* compute the 256 bit product of two 128 bit values a, b.
+   * The high 128 bits are accumulated in t and the low 128-bits
+   * in tmq. The high 128-bits of the product are returned to the
+   * address of the 1st parm. The low 128-bits are the return
+   * value.
+   */
+  const vui64_t zero = { 0, 0 };
+  vui64_t a_swap = vec_swapd ((vui64_t) a);
+  vui128_t tmh, tab, tba, tb0, tc1, tc2;
+  /* multiply the low 64-bits of a and b.  For PWR9 this is just
+   * vmsumudm with conditioned inputs.  */
+  tmq = (vui32_t) vec_vmuloud ((vui64_t)a, (vui64_t)b);
+  /* compute the 2 middle partial projects.  Use vmaddeud to add the
+   * high 64-bits of the low product to one of the middle products.
+   * This can not overflow.  */
+  tab = vec_vmuloud (a_swap, (vui64_t) b);
+  tba = vec_vmaddeud (a_swap, (vui64_t) b, (vui64_t) tmq);
+  /* sum the two middle products (plus the high 64-bits of the low
+   * product.  This will generate a carry that we need to capture.  */
+  t   = (vui32_t) vec_adduqm (tab, tba);
+  tc1 = vec_addcuq (tab, tba);
+  /* result = t[l] || tmq[l].  */
+  tmq = (vui32_t) vec_mrgald ((vui128_t) t, (vui128_t) tmq);
+  /* we can use multiply sum here because the high product plus the
+   * high sum of middle partial products can't overflow.  */
+  t   = (vui32_t) vec_permdi ((vui64_t) tc1, (vui64_t) t, 2);
+  // This is equivalent to vec_vmadd2eud(a, b, tab, tba)
+  // were (tab_even + tba_even) was pre-computed including the carry,
+  // so no masking is required.
```

Comment:
Yes but the SDK still defaults to C comments. These have been copied back from examples in the doxygen documentation which does not like C comments in examples.. The problem is that has to be done manually. Need some tool or SDK setting. Otherwise will leave this as-is for now.